src/renderers/shaders/ShaderChunk/envmap_physical_pars_fragment.glsl.js

```diff
@@ -9,7 +9,7 @@ export default /* glsl */`
 
 			vec4 envMapColor = textureCubeUV( envMap, worldNormal, 1.0 );
 
-			return PI * envMapColor.rgb * envMapIntensity;
+			return PI * envMapColor.rgb;
```

Comment:
@WestLangley Are these changes appropriate?

I'm pinging you since one can easily overlook that this PR modifies shaders in the core.

Reply:
Maybe just for additional clarification:

Before: `getIBLIrradiance`, `getIBLRadiance `and `getIBLAnisotropyRadiance` were returning the environment intensity from the map with the scaling already applied inside the function before the value is returned. This had the drawback that only global scaling per material were possible.

Now: `getIBLIrradiance`, `getIBLRadiance `and `getIBLAnisotropyRadiance` are returning the unmodified environment intensity from the map. The scaling is then applied to their returned values after they are called in lights_fragment_maps.glsl.js (this is the only place that they are ever called inside ThreeJS). This has the advantage that we can define a pixel wise scaling per material (e.g. we use it in order to let the environment illumination affect only for specular parts of a material, which was not possible before).

Reply:
>we use it in order to let the environment illumination affect only for specular parts of a material, which was not possible before

This is a physically-based shader, and the _radiance_ and _irradiance_ are jointly-consistent and scale together.

If I understand correctly, the OP wants to modify the shader so he can use Node to violate the physics of light in his app and have the environment intensity apply only to radiance.

Reply:
Thanks for the additional explanation! Against this backdrop, I see this change critical, tbh.

Reply:
@TobiasNoell Would it be an acceptable workaround to apply your shader modifications via `onBeforeCompile()`?

Reply:
The idea was to add a more generic `envMapIntensityNode` slot to the node materials (since we already have a `envMapIntensity` property for the classic materials).

It is true that the user can decide to setup a specific combination of nodes as input for `envMapIntensityNode` that would then yield a result that is not strictly correct in a physical sense, but still achieves the intended result.

In any case, it would be the user that is setting up the node. If the user decides to apply a (physically incorrect) spatially varying scaling of the environment for a material, the user should be aware of what its intentions are, right?

If the `envMapIntensityNode` is ignored or a node combination that yields any uniform floating value, everything would still remain strictly physically plausible.

Reply:
This is an example what we are trying to achieve, to make it more practical:

We have a scene that is illuminated by point lights. This gives a nice appearance including shadows on the diffuse parts of the surface, but the chrome like specular parts look too dark:
![image](https://github.com/mrdoob/three.js/assets/124624771/c7e78205-cc2d-46bd-b1d5-2c96656a954a)

To make the chrome look nicer, we setup an environment map for the scene. This makes the chrome parts look nice, but destroys the appearance of the diffuse parts:
![image](https://github.com/mrdoob/three.js/assets/124624771/532af4c2-3d68-4ecd-bc4f-df36bd024f74)

To fix this, we want to scale down the effect of the environment map for the rough parts of the surface, but this is only possible with the changes this PR proposes:
![image](https://github.com/mrdoob/three.js/assets/124624771/974a7d91-b1d6-4978-a43f-0c30420f31c8)
It is done by setting `material.envMapIntensityNode = THREENodes.pow(material.roughnessNode, THREENodes.float(4.0));`

It is true, that the last image is not strictly physically correct, but it looks the nicest out of the three 😄 

All materials in the scene are node materials. If there is any better way to achieve the same I'm happy to hear it.